src/main.js

```diff
@@ -164,10 +166,16 @@ module.exports = {
               }
             }
             let range
+            const msgLine = line - 1
+            const msgCol = column ? column - 1 : column
             try {
-              range = Helpers.rangeFromLineNumber(
-                textEditor, line - 1, column ? column - 1 : column
-              )
+              if (endColumn && endLine) {
+                validatePoint(textEditor, msgLine, msgCol)
+                validatePoint(textEditor, endLine - 1, endColumn - 1)
+                range = [[msgLine, msgCol], [endLine - 1, endColumn - 1]]
```

Comment:
Looks like _rules_ have [mixed](https://github.com/eslint/eslint/blob/0a44361d8ff319c07b92bf6c6a35890337deac0e/docs/developer-guide/working-with-rules.md#contextreport) 1-based and 0-based coords, but the results are all 1-based.